docs/howto/add_link.md

```diff
@@ -0,0 +1,336 @@
+resource.to_json
+```
+
+This would result in:
+
+```json
+{
+  "links": {
+    "home": "https://www.example.com/"
+  },
+  "data": {
+    "id": "1",
+    "type": "users",
+    "attributes": {
+      "name": "John"
+    }
+  }
+}
+```
+
+When using Rails, you can add top-level links by providing the `links` option to `render` method:
+
+```ruby
+class UserController < ActionController::Base
+  def show
+    user = User.new(id: 1, name: "John")
+    links = { home: "https://www.example.com/" }
+    render json: user, adapter: :json_api, links: links
+  end
+end
+```
+
+#### Pagination
+
+`ActiveModelSerializers` will provide automatic pagination links for collections if you use [Kaminari](https://github.com/amatsuda/kaminari)
+or [WillPaginate](https://github.com/mislav/will_paginate) and the `ActiveModelSerializers.config.jsonapi_pagination_links_enabled` config is set to `true` (defaults to `true`). To disable this feature, you can simply set this config to `false`.
+
+```ruby
+class UserController < ActionController::Base
+  def kaminari
+    users = User.all.to_a # .all.to_a used as example
+    users = Kaminari.paginate_array(page).page(3).per(1)
+    render json: users, adapter: :json_api
+  end
+
+  def will_paginate
+    users = User.all.to_a # .all.to_a used as example
+    users = users.paginate(page: 3, per_page: 1)
+    render json: users, adapter: :json_api
+  end
+end
+```
+
+Any of these actions would result in:
+```json
+{
+  "data": [
+    {
+      "type": "users",
+      "id": "3",
+      "attributes": {
+        "name": "John"
+      }
+    }
+  ],
+  "links": {
+    "self": "http://example.com/articles?page[number]=3&page[size]=1",
+    "first": "http://example.com/articles?page[number]=1&page[size]=1",
+    "prev": "http://example.com/articles?page[number]=2&page[size]=1",
+    "next": "http://example.com/articles?page[number]=4&page[size]=1",
+    "last": "http://example.com/articles?page[number]=13&page[size]=1"
+  }
+}
+```
+
+Note that it is possible to opt-out `ActiveModelSerializers` pagination relies on a collection that has the methods `current_page`, `total_pages`, and `size`, such as are supported by both [Kaminari](https://github.com/amatsuda/kaminari) or [WillPaginate](https://github.com/mislav/will_paginate). If you want to have paginated links your collections but don't want to use neither Kaminari nor WillPaginate, make sure to define these methods.
+
+### [Resource link](http://jsonapi.org/format/#document-resource-object-links)
+
+You can use the `link` class method to define the links you need in the resource's primary data:
+
+```ruby
+class UserSerializer < ActiveModel::Serializer
+  attributes :name
+
+  link(:self) { user_path(object.id) }
+  link(:posts) { posts_path(user_id: object.id) }
+end
+
+user = User.new(id: 1, name: "Kelly")
+resource = ActiveModelSerializers::SerializableResource.new(user, adapter: :json_api)
```

Comment:
this won't work without a SerializationContext.  We need to verify all our docs contain valid code.